python/tk_desktop/desktop_window.py

```diff
@@ -170,6 +180,32 @@ def __init__(self, parent=None):
 
         advanced_menu.addAction(self.ui.actionShow_Console)
 
+        # Setup the toggle action for debug logging.
+        self.toggle_debug_action = QtGui.QAction("Toggle Debug Logging", advanced_menu)
+        self.toggle_debug_action.setCheckable(True)
+        self.toggle_debug_action.setChecked(sgtk.LogManager().global_debug)
+        self.toggle_debug_action.toggled.connect(self._debug_toggled)
+
+        advanced_menu.addAction(self.toggle_debug_action)
+
+        # In the case of TK_DEBUG, we want to disable the control and set a tooltip 
+        # that explains why. We treat the TK_DEBUG env var as a complete override,
+        # otherwise we can check stored user settings. If the user previously set
+        # the logging on or off explicitly via this menu action then we'll remember
+        # it.
+        if sgtk.constants.DEBUG_LOGGING_ENV_VAR in os.environ:
```

Comment:
sgtk.constants is not part of the API. Maybe we need to introduce sgtk.LogManager().is_debug_env_var_set()? @manneohrstrom, what do you think?

Reply:
If we cannot use `sgtk.LogManager().global_debug`, adding a method to core sounds useful.